lib/internal/util.js

```diff
@@ -41,7 +45,10 @@ function deprecate(fn, msg, code) {
     if (!warned) {
       warned = true;
       if (code !== undefined) {
-        process.emitWarning(msg, 'DeprecationWarning', code, deprecated);
+        if (!codesWarned[code]) {
+          process.emitWarning(msg, 'DeprecationWarning', code, deprecated);
+          codesWarned[code] = true;
+        }
```

Comment:
This should be obsolete due to the `warned` variable. It should actually do exactly this.

Reply:
@BridgeAR No, `warned` doesn't take care of this situation. `warned` is scoped to the function so this will emit twice currently:

```js
{
  const msg = 'cannonmouth';
  const code = 'deprecatesque';
  const fn1 = util.deprecate(() => {}, msg, code);
  const fn2 = util.deprecate(() => {}, msg, code);
  expectedWarnings.set(msg, { code, count: 1 });
  fn1();
  fn2();
```

Reply:
That is correct but normally the deprecation function is only called once with a message. We could remove the `warned` variable though as it got obsolete with `codesWarned`.

Reply:
@BridgeAR I thought about removing `warned` but it's still used if `code` is not provided.